frontend/app.js

```diff
@@ -0,0 +1,510 @@
+            taskName={this.props.taskName}
+            taskDescription={this.props.taskDescription}
+          />
+        </div>
+        <div className='wrapper-list'>
+          <Task_list
+            tasks={this.props.tasks}
+            completeTask={this.props.completeTask}
+            removeTask={this.props.removeTask}
+            completedState={this.props.completedState}
+          />
+        </div>
+      </div>
+    )
+  }
+}
+
+class Task_list extends React.Component {
+  constructor() {
+    super()
+
+    this.state = {
+      taskFocus: 0,
+      name: null,
+      description: null
+    }
+  }
+
+  focusOnTask(event) {
+    console.log('/task/' + event)
+    if (this.state.taskFocus === 0) {
+      fetch('/task/' + event)
+       .then((data) => {
+        return data.json()
+      }).then((json) => {
+        this.setState({
+          taskFocus: 1,
+          name: json.name,
+          description: json.description,
+          id: json.id
+        })
+      })
+
+    } else if (this.state.taskFocus === 1) {
+      this.setState({
+        taskFocus: 0
+      })
+    }
+  }
+
+  render() {
+      if (this.state.taskFocus === 0) {
+        return(
+          <ul className='list' >
+            {this.props.tasks
+              .filter(task => task.completed === 0)
+              .map((task, i) =>
+                <Task
+                  key={i}
+                  task={task.name}
+                  description={task.description}
+                  id={task.id}
+                  taskFocus={this.state.taskFocus}
+                  onCompleteTask={this.props.completeTask.bind(this, task.id)}
+                  onRemoveTask={this.props.removeTask.bind(this, task.id)}
+                  onSelect={this.focusOnTask.bind(this, task.id)}
+                />)
+            }
+          </ul>
+        )
+      } else if (this.state.taskFocus === 1) {
+        return(
+          <div>
+            <ul className='list' >
+              {this.props.tasks
+                .filter(task => task.completed === 0)
+                .map((task, i) =>
+                  <Task
+                    key={i}
+                    task={task.name}
+                    description={task.description}
+                    id={task.id}
+                    taskFocus={this.state.taskFocus}
+                    onCompleteTask={this.props.completeTask.bind(this, task.id)}
+                    onRemoveTask={this.props.removeTask.bind(this, task.id)}
+                    onSelect={this.focusOnTask.bind(this, task.id)}
+                  />)
+              }
+            </ul>
+            <div>
+              <Focused_task onSelect={this.focusOnTask.bind(this)} taskFocus={this.state.taskFocus} name={this.state.name} description={this.state.description} id={this.state.id} onCompleteTask={this.props.completeTask.bind(this, this.state.id)} />
+            </div>
+          </div>
+        )
+      }
+  }
+}
+
+class Task extends React.Component {
```

Comment:
You could use a pure function here. You'd prefer these over class components whenever possible. This is what we'd call a presentational component.

```
const Task = () => (
  <div className='task'>
  </div>
);
```